### PR TITLE
Ability to disable Identify for a layer #6101

### PR DESCRIPTION
**Pull request by myrdstom**
## Description

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [x] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

#6101

**What is the new behavior?**
Screenshot 2020-11-06 at 15 25 24
Screenshot 2020-11-06 at 15 25 36

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

## Other useful information
This PR is a cleanup of the [following](https://github.com/geosolutions-it/MapStore2/pull/6128) PR

### CODE REVIEW

File: web/client/components/data/identify/viewers/HiddenViewer.jsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+            <div id="container">
+                <Alert bsStyle={"danger"}>
+                    <h4><HTML msgId="noFeatureInfo"/></h4>
+                </Alert>
+            </div>
+        );
+    }
+}
+
+module.exports = HiddenViewer;
```
Comment (offtherailz) [inline web/client/components/data/identify/viewers/HiddenViewer.jsx:19]: The hidden viewer and it's test is not needed anymore

Comment (myrdstom) [reply web/client/components/data/identify/viewers/HiddenViewer.jsx:19]: When looking at option 2) of the GitHub(#6101) issue. I use the Hidden viewer to render the error message Alert when calling `getViewers()` in MapInfoUtils. How would I go around this?

**Review by offtherailz — CHANGES_REQUESTED** [review]
This solution introduces a lot things that are not necessary, and hard to maintain. Moreover this way the request is performed anyway, and we want to avoid it. 
- [ ] As I said in my previous review if you  set the `queryable` flag for the layer to "false", frome the UI, the filtering is already applied from the selectors, and you don't need to operate to anything else in epics.
https://github.com/geosolutions-it/MapStore2/blob/dbe79959a9868907827fdf75f3cb94369138a27c/web/client/utils/MapInfoUtils.js#L213
- If you are not sure to use the queryable flag, anyway you can filter layers with the format in the same filter function and the request will not be performed anymore.
- [ ] please remove all things not needed anymore
- [ ] please test only the remaining changes.

File: web/client/components/data/identify/viewers/HiddenViewer.jsx (hunk 1 above)
Comment (offtherailz) [reply web/client/components/data/identify/viewers/HiddenViewer.jsx:19]: When layers are filtered on the beginning (excluding not visible, not queriable), this happens automatically.
As you can see if I click and the layer is not visible, the error message appears.
image

Comment (offtherailz) [reply web/client/components/data/identify/viewers/HiddenViewer.jsx:19]: probably this was the misundersanding :smile:

Comment (myrdstom) [reply web/client/components/data/identify/viewers/HiddenViewer.jsx]: Screenshot 2020-11-06 at 09 56 05
This is what it should be, correct?

File: web/client/components/data/identify/__tests__/DefaultViewer-test.jsx [hunk 2]
```diff
@@ -53,6 +53,11 @@ describe('DefaultViewer', () => {
             layerMetadata: {
                 title: 'a'
             }
+        }, {
```
Comment (offtherailz) [inline web/client/components/data/identify/__tests__/DefaultViewer-test.jsx:56]: these tests changes are not needed anymore, for the whole file, you are not modifying this anymore

File: web/client/utils/__tests__/MapInfoUtils-test.js [hunk 3]
```diff
@@ -53,7 +53,7 @@ describe('MapInfoUtils', () => {
     });
 
     it('getAvailableInfoFormatLabels', () => {
-        let testData = ['TEXT', 'PROPERTIES', 'HTML', 'TEMPLATE'];
+        let testData = ['TEXT', 'PROPERTIES', 'HTML', 'TEMPLATE', 'HIDDEN'];
```
Comment (offtherailz) [inline web/client/utils/__tests__/MapInfoUtils-test.js:56]: this is not needed
```suggestion
        let testData = ['TEXT', 'PROPERTIES', 'HTML', 'TEMPLATE'];
```

File: web/client/utils/MapInfoUtils.js [hunk 4]
```diff
@@ -211,7 +212,7 @@ const MapInfoUtils = {
         return l.visibility &&
             MapInfoUtils.services[l.type] &&
             (l.queryable === undefined || l.queryable) &&
-            l.group !== "background"
+            l.group !== "background" && l?.featureInfo?.format !== 'HIDDEN'
```
Comment (offtherailz) [inline web/client/utils/MapInfoUtils.js:215]: This function should be tested to verify that a layer with featureinfo.format === 'hidden' returns false.

**Review by offtherailz — CHANGES_REQUESTED** [review]
Almost there :+1: please clean up your changes and restore the tests of components you don't modify anymore, test your part instead.

File: web/client/components/data/identify/__tests__/DefaultViewer-test.jsx (hunk 2 above)
Comment (myrdstom) [reply web/client/components/data/identify/__tests__/DefaultViewer-test.jsx:56]: true, Im going to revert all my changes to this file

**Review by offtherailz — APPROVED** [review]

**Pull request merged**